test/runtests.jl

```diff
@@ -3,41 +3,63 @@ import GeoFormatTypes as GFT
 import GeoInterface as GI
 using Test
 import ArchGDAL
+import LibGEOS
 
 @testset "WellKnownGeometry.jl" begin
 
     coord = [1.1, 2.2]
+    coord3 = [1.1, 2.2, 3.3]
     lcoord = [3.3, 4.4]
-    ring = [[0.1, 0.2], [0.3, 0.4], [0.1, 0.2]]
-    coords = [[coord, lcoord, coord], ring]
-
-    for (type, geom) in (
-        ("Point", ArchGDAL.createpoint(coord)),
-        ("LineString", ArchGDAL.createlinestring([coord, lcoord])),
-        ("Polygon", ArchGDAL.createpolygon(coords)),
-        ("MultiPoint", ArchGDAL.createmultipoint([coord, lcoord])),
-        ("MultiLineString", ArchGDAL.createmultilinestring([[coord, lcoord], [coord, lcoord]])),
-        ("MultiPolygon", ArchGDAL.createmultipolygon([coords, coords])),
-        ("Empty", ArchGDAL.createpoint()),
-        ("Empty Multi", ArchGDAL.createmultipolygon())
+    lcoord3 = [3.3, 4.4, 5.5]
+    ring = [(0.1, 0.2), (0.3, 0.4), (0.1, 0.2)]
+    ring3 = [(0.1, 0.2, 0.3), (0.3, 0.4, 0.5), (0.1, 0.2, 0.3)]
+    coords = [[Tuple(coord), Tuple(lcoord), Tuple(coord)], ring]
+    coords3 = [[Tuple(coord3), Tuple(lcoord3), Tuple(coord3)], ring3]
+
+    for (type, geom, broken, threed) in (
+        ("Point", ArchGDAL.createpoint(coord), false, false),
+        ("PointZ", ArchGDAL.createpoint(coord3), true, true),
+        ("LineString", ArchGDAL.createlinestring(coord, lcoord), false, false),
+        ("LineStringZ", ArchGDAL.createlinestring(coord3, lcoord3, lcoord3), true, true),
+        ("Polygon", ArchGDAL.createpolygon(coords), false, false),
+        ("PolygonZ", ArchGDAL.createpolygon(coords3), true, true),
+        ("MultiPoint", ArchGDAL.createmultipoint([coord, lcoord]), false, false),
+        ("MultiPointZ", ArchGDAL.createmultipoint([coord3, lcoord3]), true, true),
+        ("MultiLineString", ArchGDAL.createmultilinestring([[coord, lcoord], [coord, lcoord]]), false, false),
+        ("MultiLineStringZ", ArchGDAL.createmultilinestring([[coord3, lcoord3], [coord3, lcoord3]]), true, true),
+        ("MultiPolygon", ArchGDAL.createmultipolygon([coords, coords]), false, false),
+        ("MultiPolygonZ", ArchGDAL.createmultipolygon([coords3, coords3]), true, true),
+        ("Empty", ArchGDAL.createpoint(), false, false),
+        ("Empty Multi", ArchGDAL.createmultipolygon(), false, false)
```

Comment:
Could these be replaced by GeoInterface wrapper types, since they encode directly whether Z and M are specifically present?

Reply:
They could be, but I mostly want to compare to GDAL and GEOS WKB/WKT functionality, so that would only introduce a layer of conversions.